logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/call/usecase/EndOngoingCallManager.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.feature.call.usecase
+
+import com.wire.kalium.logic.data.id.ConversationId
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+
+interface EndOngoingCallManager {
+    suspend fun observeCallEndedBecauseOfVerificationDegraded(): Flow<ConversationId>
+    suspend fun onCallEndedBecauseOfVerificationDegraded(conversationId: ConversationId)
+}
```

Comment:
I'd suggest to rename this class as we don't manage anything here we are just observing. How about `EndCallResultListener`, `EndCallResultReceiver` ?

Reply:
yeah, i like listener